##### Q1. What are the three measures of central tendency?

Ans- The three measures of central tendency are Mean , Mode, and the Median.

1. Mean - The sum of all the values divided by the total number of values.It is also called the arithmetic mean.

2. Mode - The most frequent values in the dataset.

3. Median - The middle value in an ordered dataset. If the dataset has an even number of values, the median is the mean of the middle two values.

##### Q2. What is the difference between the mean, median, and mode? How are they used to measure the central tendency of a dataset

Ans - The mean is the average of all the values in the dataset. It is calculated by adding up all the values and dividing by the number of values. The mean is sensitive to outliers or extreme values in the dataset, which can make it higher or lower than the true centre of the data.

The median is the middle value in an ordered dataset. It is calculated by sorting the values from lowest to highest and finding the value that splits the dataset in half. If there are an even number of values, the median is the average of the middle two values. The median is not affected by outliers or extreme values, so it can be a better measure of central tendency for skewed or non-normal distributions.

The mode is the most frequent value in the dataset. It is calculated by counting how many times each value occurs and finding the value that occurs most often. There can be more than one mode or no mode at all, depending on how many values have the same frequency. The mode can be used for any type of data, including categorical or nominal data, such as colors, names, or labels.

##### Q3. Measure the three measures of central tendency for the given height data:
[178,177,176,177,178.2,178,175,179,180,175,178.9,176.2,177,172.5,178,176.5]

In [1]:
import numpy as np
s = np.array([178,177,176,177,178.2,178,175,179,180,175,178.9,176.2,177,172.5,178,176.5])

In [2]:
np.mean(s)

177.01875

In [3]:
np.median(s)

177.0

In [4]:
from scipy import stats
stats.mode(s , keepdims = False)

ModeResult(mode=177.0, count=3)

##### Q4. Find the standard deviation for the given data:
[178,177,176,177,178.2,178,175,179,180,175,178.9,176.2,177,172.5,178,176.5]

In [5]:
s

array([178. , 177. , 176. , 177. , 178.2, 178. , 175. , 179. , 180. ,
       175. , 178.9, 176.2, 177. , 172.5, 178. , 176.5])

In [6]:
np.std(s)

1.7885814036548633

##### Q5. How are measures of dispersion such as range, variance, and standard deviation used to describe the spread of a dataset? Provide an example.

Ans - Measures of dispersion are statistical tools that describe how spread out or varied a dataset is. They help to understand the distribution of the data and the degree of variability or heterogeneity among the data values.

1. Range: The difference between the maximum and minimum values in a dataset. For example, if a dataset has values 2, 4, 6, 8, 10, then the range is 10 - 2 = 8.

2. Variance: The average of the squared deviations of each data value from the mean. For example, if a dataset has values 1, 2, 3, 4, 5, then the mean is 3 and the variance is [(1−3)**2+(2−3)**2+(3−3)**2+(4−3)**2+(5−3)**2]/5 =2
.
3. Standard deviation: It is the square root of the variance. It measures how much the data values deviate from the mean on average. For example, if a dataset has a variance of 2, then the standard deviation is 1.414


In [7]:
import seaborn as sns

In [8]:
df = sns.load_dataset("flights")
df.head()

year month  passengers
0  1949   Jan         112
1  1949   Feb         118
2  1949   Mar         132
3  1949   Apr         129
4  1949   May         121

In [18]:
rng = df.passengers.max()-df.passengers.min()
print("Mean = ",df.passengers.mean())
print("Variance = ",df.passengers.var())
print("Standard Deviation = ",df.passengers.std())
print("Range = ",rng)

Mean =  280.2986111111111
Variance =  14391.917200854701
Standard Deviation =  119.96631694294321
Range =  518


This means that on average, the monthly totals of airline passengers deviate by about 120 passengers from their mean, which is about 280 passengers. The range of passengers is 518, which means that there is a difference of 518 passengers between the lowest and highest monthly totals.

In [19]:
## also we can call describe function 
df.describe()

year  passengers
count   144.000000  144.000000
mean   1954.500000  280.298611
std       3.464102  119.966317
min    1949.000000  104.000000
25%    1951.750000  180.000000
50%    1954.500000  265.500000
75%    1957.250000  360.500000
max    1960.000000  622.000000

##### Q6. What is a Venn diagram?

Ans - Venn diagrams are diagrams that show the relationships between sets of items or concepts using circles or other shapes. They can help to visualize the similarities and differences, the intersections and unions, and the probabilities and logical operations involving sets.

Uses:

Exploring and comparing categorical variables in a dataset, such as gender, age group, education level, etc.

Illustrating the results of set operations, such as joins, merges, filters, etc., on data tables or data frames.

Communicating and presenting complex data analysis in a clear and intuitive way.


##### Q7. For the two given sets A = (2,3,4,5,6,7) & B = (0,2,6,8,10). Find:

(i) 	A B

(ii)	A ⋃ B

In [30]:
A = {2,3,4,5,6,7}
B = {0,2,6,8,10}

In [36]:
# finding Union of two sets using | operator
C = A|B
C

{0, 2, 3, 4, 5, 6, 7, 8, 10}

In [37]:
## finding intersection using & operator
D = A&B
D

{2, 6}

##### Q8. What do you understand about skewness in data?

Ans - Skewness in data is a measure of how asymmetric or distorted a distribution is from a symmetrical or normal distribution. Skewness can be positive or negative, depending on whether the data is skewed to the right or to the left of the median. Skewness affects the mean, median, and mode of the data and can have implications for statistical inference and modeling.

A normal distribution has zero skew and is symmetrical on both sides of the center point. A positively skewed distribution has a long tail on its right side and a negatively skewed distribution has a long tail on its left side. In a positively skewed distribution, the mean is greater than the median and median is greater than mode, and in a negatively skewed distribution, the mean is less than the median and median is less than mode.

##### Q9. If a data is right skewed then what will be the position of median with respect to mean?

Ans - If a data is right skewed, then the median will be less than the mean. This is because the mean is pulled to the right by the long tail of the distribution, while the median is the middle value that divides the data into two equal halves.

##### Q10. Explain the difference between covariance and correlation. How are these measures used in  statistical analysis?

Ans - Covariance and correlation are two ways to measure how two variables change together. Covariance shows the direction of the relationship, while correlation shows the direction and the strength of the relationship. These measures are used in statistical analysis to test hypotheses, compare results, and check assumptions.

Covariance and correlation can also be used to assess the validity of regression models, by checking the assumptions of linearity, homoscedasticity, and multicollinearity.

##### Q11. What is the formula for calculating the sample mean? Provide an example calculation for a  dataset.

Ans - The formula for calculating the sample mean is:
$\bar{x} = \frac{1}{n} \sum_{i=1}^n x_i$
where:

xˉ is the sample mean
n is the sample size
xi​ is the value of the i-th observation in the sample
∑ means the sum of all values

In [38]:
df

year month  passengers
0    1949   Jan         112
1    1949   Feb         118
2    1949   Mar         132
3    1949   Apr         129
4    1949   May         121
..    ...   ...         ...
139  1960   Aug         606
140  1960   Sep         508
141  1960   Oct         461
142  1960   Nov         390
143  1960   Dec         432

[144 rows x 3 columns]

In [46]:
sum = df["passengers"].sum()
sum

40363

In [50]:
n = len(df["passengers"])
n

144

In [51]:
sample_mean = sum/n
sample_mean

280.2986111111111

In [52]:
## orsimply we can use 
df.passengers.mean()

280.2986111111111

##### Q12. For a normal distribution data what is the relationship between its measure of central tendency?

Ans - For a normal distribution data, the relationship between its measure of central tendency is that the mean, the median, and the mode are all equal. This means that the distribution is symmetrical and has a single peak at the center.

##### Q13. How is covariance different from correlation?

Ans - Covariance indicates the direction of the relationship, while correlation indicates both the direction and the strength of the relationship.

Covariance is not standardized, while correlation is standardized. This means that covariance depends on the units and scales of the variables, while correlation does not. Correlation ranges from -1 to 1, while covariance can take any value.


##### Q14. How do outliers affect measures of central tendency and dispersion? Provide an example.

Ans - Outliers are extreme values that deviate significantly from the rest of the data. They can affect the measures of central tendency and dispersion in different ways, depending on the type of measure.

1. Measures of central tendency, such as the mean, the median, and the mode, are used to describe the typical or representative value of a data set. Outliers can affect these measures by pulling them towards the direction of the outlier, making them less representative of the majority of the data.

For example, if we have a data set of 10 salaries: 40, 45, 50, 55, 60, 65, 70, 75, 80, and 500 (in thousands), the mean salary is 104, which is influenced by the outlier of 500. The median salary is 62.5, which is less affected by the outlier. The mode salary is not affected by the outlier, as it is the most frequent value in the data (there is no mode in this case).

2. Measures of dispersion, such as the range, the variance, and the standard deviation, are used to describe how spread out or variable the data are. Outliers can affect these measures by increasing them, making them reflect more variability than there actually is in the majority of the data.

For example, using the same data set of salaries as above, the range is 460 (500 - 40), which is influenced by the outlier of 500. The variance is 18136, which is also influenced by the outlier. The standard deviation is 134.67, which is also affected by the outlier.